src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -45,12 +59,27 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
     }
 
+    private void setTargetIndex(List<Person> lastShownList) throws CommandException {
+        int temp = lastShownList.stream()
+                .filter(person -> person.getName().equalsIgnoreCase(targetName))
+                .map(lastShownList::indexOf)
+                .reduce(-1, (x, y) -> y);
+        if (temp == -1) {
+            throw new CommandException(Messages.MESSAGE_PERSON_NOT_IN_ADDRESS_BOOK);
+        }
+        this.targetIndex = Index.fromZeroBased(temp);
+    }
+
     @Override
     public boolean equals(Object other) {
         if (other == this) {
             return true;
         }
 
+        if (targetIndex == null) {
+            return false;
+        }
```

Comment:
If the `other` command also has a null `targetIndex`, would there be a chance that they are actually equal?

Maybe this could be changed to something like:
```java
if (targetIndex == null) {
   if (other.targetIndex == null) {
      if (targetName == null) {
          returns other.targetName == null; 
      }
      return targetName.equals(other.targetName);
   }
   return false;
}
```
**after line 88 which casts the `Object` to `DeleteCommand`**.
Note that this does not check if they are deleting the same person, it only checks if both name and index are equal.

Reply:
Good catch!